### PR TITLE
Update to `7.3.0b2`

### PR DESCRIPTION
**Pull request by jtpio**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jtpio** [comment]
@conda-forge-admin, please rerender

**Comment by jtpio** [comment]
@conda-forge-admin, please rerender

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {% set version = "7.3.0b2" %}
 
+{% set python_min = python_min | default("3.9") %}
```
Comment (jtpio) [inline recipe/meta.yaml:3]: Upstream has not dropped `3.8` yet: https://github.com/jupyter/notebook/blob/606190117a34f9488bbdf87d90feb219755a6153/pyproject.toml#L10

But I guess it's fine to do it here already?

Comment (bollwyvl) [reply recipe/meta.yaml:3]: Right, but `conda-forge` _has_ already dropped the EOL python: https://github.com/conda-forge/cfep/blob/main/cfep-25.md

**Review by bollwyvl — APPROVED** [review]

**Comment by jtpio** [comment]
Thanks @bollwyvl for the help!

**Pull request merged**